### ISSUE TITLE
SQLite - `COLLATE NOCASE` causes SQLFluff to fail SQL parsing

### ISSUE THREAD

**Issue by joeljuca** [sqlfluff/sqlfluff#4829]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using `COLLATE NOCASE` in SQLite tables to make `TEXT` columns case-insensitive, SQLFluff fails to parse the whole `CREATE TABLE` block.

Given two SQLite-specific SQL files:

```sql
-- create-users.sql
create table users (
  username text not null
);
```

```sql
-- create-users-nocase.sql
);
create table users (
  username text not null collate nocase
);
```

Running SQLFluff against them gives me the following results:

```
(.venv)  $ sqlfluff lint ./create-user.sql
All Finished 📜 🎉!
```

```
(.venv)  $ sqlfluff lint ./create-users-nocase.sql
== [create-users-nocase.sql] FAIL
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'create
                       | table users (\n  username text not...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

SQLFluff should be able to parse SQLite-specific `COLLATE NOCASE` clauses.

### Observed Behaviour

The SQLite dialect does not understand SQLite COLLATE clauses. https://www.sqlite.org/datatype3.html#collation

### How to reproduce

Given two SQLite-specific SQL files:

```sql
-- create-users.sql
create table users (
  username text not null
);
```

```sql
-- create-users-nocase.sql
);
create table users (
  username text not null collate nocase
);
```

Running SQLFluff against them gives me the following results:

```
(.venv)  $ sqlfluff lint ./create-user.sql
All Finished 📜 🎉!
```

```
(.venv)  $ sqlfluff lint ./create-users-nocase.sql
== [create-users-nocase.sql] FAIL
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'create
                       | table users (\n  username text not...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

SQLite

### Version

2.0.7

### Configuration

```ini
; SQLFluff config file
; See https://docs.sqlfluff.com/en/stable/configuration.html#defaultconfig

[sqlfluff]
dialect = sqlite
templater = raw

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joeljuca** [comment]
I'm not experienced in Python, so I can't submit a PR for it. :(
